lightning_uq_box/uq_methods/laplace_model.py

```diff
@@ -238,22 +227,100 @@ class LaplaceRegression(LaplaceBase):
+                for `pred_type='nn'` only 'mc' is supported
+            num_samples: number of samples for prediction, if specified
+                will call `predictive_samples` instead of `predictive` method in
+                Laplace library
+            tune_prior_precision_and_sigma: whether to tune prior precision and sigma
+            tuning_lr: learning rate for tuning prior precision and sigma
+            n_epochs_tuning: number of epochs to tune prior precision and sigma
         """
-        super().__init__(laplace_model)
+        super().__init__(laplace_model, pred_type, link_approx, num_samples)
 
         assert self.laplace_model.likelihood == "regression"
 
         self.loss_fn = torch.nn.MSELoss()
 
+        self.tuning_lr = tuning_lr
+        self.n_epochs_tuning = n_epochs_tuning
+        self.tune_prior_precision_and_sigma = tune_prior_precision_and_sigma
+
+    def on_test_start(self) -> None:
+        """Fit the Laplace approximation before testing."""
+        super().on_test_start()
+        if self.tune_prior_precision_and_sigma:
+            tune_prior_precision(
+                self.laplace_model, self.tuning_lr, self.n_epochs_tuning
+            )
+
     def setup_task(self) -> None:
         """Set up task specific attributes."""
         self.test_metrics = default_regression_metrics("test")
 
+    def forward(self, X: Tensor, **kwargs: Any) -> dict[str, Tensor]:
+        """Fitted Laplace Model Forward Pass.
+
+        Args:
+            X: tensor of data to run through the model [batch_size, input_dim]
+            kwargs: additional arguments for laplace forward pass
+
+        Returns:
+            output from the laplace model
+        """
+        if not self.laplace_fitted:
+            self.on_test_start()
+
+        pred_dict: dict[str, Tensor] = {}
+        if self.num_samples:
+            fsamples = self.laplace_model.predictive_samples(
+                X, pred_type=self.pred_type, n_samples=self.num_samples
+            )
+            mean = fsamples.mean(0).squeeze()
+
+            # return samples as shape [batch_size, out_dim, num_samples]
+            pred_dict["samples"] = fsamples.permute(1, 2, 0)
+            laplace_epistemic = fsamples.std(0).squeeze()
+            laplace_aleatoric = (
+                torch.ones_like(laplace_epistemic)
+                * self.laplace_model.sigma_noise.item()
+            )
+            pred_std = torch.sqrt(laplace_epistemic + laplace_aleatoric**2)
+        else:
+            mean, var = self.laplace_model(
+                X, pred_type=self.pred_type, link_approx=self.link_approx
+            )
+            mean = mean.squeeze().detach()
+            laplace_epistemic = var.squeeze().sqrt()
+
+            laplace_aleatoric = (
+                torch.ones_like(laplace_epistemic)
+                * self.laplace_model.sigma_noise.item()
+            )
+            pred_std = torch.sqrt(laplace_epistemic**2 + laplace_aleatoric**2)
```

Comment:
That's a good point. Actually, all methods we have implemented at the moment are just tested for 1D Regression tasks. I think training for a majority of methods will work, however, it will break in a few places, certainly for logging. Here, it will break if people want to do joint predictions (available in Laplace with `join=True`) argument because the shapes will change. Thus I think, extension to N-D regression cases is worth a separate PR.

Reply:
Yes that was our rational as well. Where do you think such a note or "disclaimer" would be best placed? On the docs landing page and README or in the tutorial notebooks?

Reply:
The docs landing page actually already says


The library currently supports the following four tasks: 

*  Regression for tabular/image inputs with 1D scalar targets
* ...

I just missed that.